include/mbgl/map/query.hpp

```diff
@@ -6,13 +6,24 @@
 namespace mbgl {
 
 /**
- * Options for Map queries.
+ * Options for query rendered features.
  */
-class QueryOptions {
+class RenderedQueryOptions {
 public:
     /** layerIDs to include in the query */
     optional<std::vector<std::string>> layerIDs;
-    
+
+    optional<style::Filter> filter;
+};
+
+/**
+ * Options for query source features
+ */
+class SourceQueryOptions {
+public:
+    // Required for VectorSource, ignored for GeoJSONSource
+    optional<std::string> sourceLayer;
```

Comment:
Let's make this a `std::vector<std::string>` like `RenderedQueryOptions`. I don't think there's a good reason that the GL JS API is inconsistent (https://github.com/mapbox/mapbox-gl-js/issues/4372).